### DISCUSSION TITLE
Commit message differs after applying a patch

### DISCUSSION THREAD

**Discussion by douniii** [git-for-windows/git#3182] [Q&A] [answered]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
```
$ git --version --build-options

git version 2.31.1.windows.1
cpu: x86_64
built from commit: c5f0be26a7e3846e3b6268d1c6c4800d838c6bbb
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon

```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?
```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19041.928]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path: 
Default Branch Option:  
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Core
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

None

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

In a empty folder:
```
git init
touch test.txt
git add *
git commit -m '[core] Adds new file'
git format-patch -M -1
```
Create another empty folder and copy the generated patch file in it
```
git init
git am *.patch
git log
```
 - What did you expect to occur after running these commands?

In the second repository, in which the patch was applied, the commit message should be
`[core] Adds new file`, like in the first repository.

 - What actually happened instead?

The commit message is `Adds new file`. So `[core] ` was removed.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

No special repo.

**Comment by PhilipOakley** [comment]
Is this not an artefact of the `[PATCH ..]` application process where the initial `[abcd...] ` is stripped from the patch subject?

**Comment by rimrul** [comment, marked as answer]
This is intentional behaviour. See `git am --keep` and `git am --keep-non-patch`.

**Reply by bryanemmanuelbasor** [reply]
$ gpg --list-secret-keys --keyid-format=long
/Users/hubot/.gnupg/secring.gpg
------------------------------------
sec   4096R/3AA5C34371567BD2 2016-03-10 [expires: 2017-03-10]
uid                          Hubot <hubot@example.com>
ssb   4096R/4BB6D45482678BE3 2016-03-10
